### PR TITLE
[NFC][ADT] Add range wrapper for std::mismatch

### PR DESCRIPTION
**Pull request by jurahul**

### CODE REVIEW

File: llvm/include/llvm/ADT/STLExtras.h [hunk 1]
```diff
@@ -2002,6 +2002,13 @@ template <typename R, typename Compare> auto max_element(R &&Range, Compare C) {
   return std::max_element(adl_begin(Range), adl_end(Range), C);
 }
 
+/// Provide wrappers to std::mismatch which take ranges instead of having to
+/// pass begin/end explicitly.
+template <typename R1, typename R2> auto mismatch(R1 &&Range1, R2 &&Range2) {
```
Comment (kuhar) [inline llvm/include/llvm/ADT/STLExtras.h:2007]: Could you add a comment of explanation for what this algorithm does? (and maybe an example too) I think that this is a lesser known std function

Comment (kuhar) [reply llvm/include/llvm/ADT/STLExtras.h:2007]: Also, do we want to support the overload that takes a binary predicate?

Comment (jurahul) [reply llvm/include/llvm/ADT/STLExtras.h:2007]: I don't intend to use that overload. My understanding is that they are added on-demand. But I can add it.

Comment (kuhar) [reply llvm/include/llvm/ADT/STLExtras.h:2007]: I think it's fine either way

Comment (jurahul) [reply llvm/include/llvm/ADT/STLExtras.h]: SG, then I'll just add what I intend to use. I added a comment with an example. PTAL.

**Review by kuhar — APPROVED** [review]
LGTM

**Comment by jurahul** [comment]
Thanks.

**Comment by dwblaikie** [comment]
(FWIW, I /mildly/ prefer these wrappers to have test coverage - but not a huge deal)

**Comment by jurahul** [comment]
Thanks. I see there is a STLExtrasTest.cpp, I'll add a unit test for maximal there.

**Pull request merged**